src/main/kotlin/org/stellar/walletsdk/Anchor.kt

```diff
@@ -0,0 +1,185 @@
+package org.stellar.walletsdk
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.stellar.sdk.Network
+import org.stellar.sdk.Server
+import org.stellar.walletsdk.util.GsonUtils
+import org.stellar.walletsdk.util.StellarToml
+import org.stellar.walletsdk.util.interactiveFlow
+
+/**
+ * Build on/off ramps with anchors.
+ *
+ * @property server Horizon [Server] instance
+ * @property network Stellar [Network] instance
+ * @property httpClient optional custom HTTP client, uses [OkHttpClient] by default
+ */
+class Anchor(
+  private val server: Server,
+  private val network: Network,
+  private val httpClient: OkHttpClient = OkHttpClient()
+) {
+  /**
+   * Get anchor information from a TOML file.
+   *
+   * @param homeDomain home domain where to find stellar.toml file
+   *
+   * @return TOML file content
+   */
+  suspend fun getInfo(homeDomain: String): Map<String, Any> {
+    return CoroutineScope(Dispatchers.IO)
+      .async {
+        val toml = StellarToml(homeDomain, server, httpClient)
+
+        return@async toml.getToml()
+      }
+      .await()
+  }
+
+  /**
+   * Available anchor services and information about them. For example, limits, currency, fees,
+   * payment methods.
+   *
+   * @param serviceUrl URL where `/info` endpoint is hosted
+   *
+   * @return a list of available anchor services
+   *
+   * @throws [NetworkRequestFailedException] if network request fails
+   * @throws [InvalidAnchorServiceUrl] if provided service URL is not a valid URL
+   */
+  suspend fun getServicesInfo(serviceUrl: String): AnchorServiceInfo {
+    val url: HttpUrl
+
+    try {
+      url = serviceUrl.toHttpUrl()
+    } catch (e: Exception) {
+      throw InvalidAnchorServiceUrl(e)
+    }
+
+    val urlBuilder = HttpUrl.Builder().scheme(url.scheme).host(url.host).port(url.port)
+
+    url.pathSegments.forEach { urlBuilder.addPathSegment(it) }
+    urlBuilder.addPathSegment("info")
+
+    val infoUrl = urlBuilder.build().toString()
+
+    val request = Request.Builder().url(infoUrl).build()
+    val gson = GsonUtils.instance!!
+
+    return CoroutineScope(Dispatchers.IO)
+      .async {
+        httpClient.newCall(request).execute().use { response ->
+          if (!response.isSuccessful) throw NetworkRequestFailedException(response)
+
+          val infoResponse = response.body!!.charStream()
+          return@async gson.fromJson(infoResponse, AnchorServiceInfo::class.java)
```

Comment:
Something to keep in mind here is that right now, you only have one `AnchorServiceInfo` class. But when the Wallet SDK supports SEP-6, 31, or 38, we'll have more.

That means that this function may need to attempt to deserialize the same response into a few different objects. Maybe that is acceptable, but it is inefficient. Ideally, we were able to know which object to use.

Reply:
Yes, we will need to use something generic here that can be passed when called.

Reply:
@JakeUrban in future we may want to use polymorphic de-serialization, think it's a good TODO.